File: python/src/common/lookandsay.py
def get_length_of_sequence(sequence_number: int) -> int:
    return len(get_sequence(sequence_number))


def get_sequence(sequence_number: int) -> str:
    sequence = [1]
    for _ in range(sequence_number - 1):
        temporary = []
        c = sequence[0]
        count = 0

        def store():
            temporary.append(count)
            temporary.append(c)

        for seq_char in sequence:
            if c != seq_char:
                store()
                c = seq_char
                count = 0
            count += 1

        store()
        sequence = temporary

    return ''.join(map(str, sequence))

File: cpp/src/look_and_say/common/look_and_say.cpp
#include "look_and_say.hpp"

#include <vector>


namespace look_and_say::common
{
uint64_t getLengthOfSequence(uint32_t sequence_number)
{
    return getSequence(sequence_number).length();
}

std::string getSequence(uint32_t sequence_number)
{
    if (sequence_number == 0)
        return "";

    std::vector<uint8_t> sequence{1};

    while (--sequence_number > 0)
    {
        std::vector<uint8_t> temporary;

        uint8_t c = sequence[0];
        uint8_t count = 0;

        auto store = [&c, &count, &temporary]() {
          temporary.push_back(count);
          temporary.push_back(c);
        };

        for (size_t i = 0; i < sequence.size(); ++i, ++count)
        {
            if (c != sequence[i])
            {
                store();
                c = sequence[i];
                count = 0;
            }
        }

        store();
        std::swap(sequence, temporary);
    }

    std::string seq;
    for (auto const item : sequence)
        seq.append(std::to_string(item));

    return seq;
}
} // namespace look_and_say::common
